daemon/src/db.rs

```diff
@@ -472,6 +473,56 @@ mod tests {
         assert_eq!(cfd, cfd_from_db)
     }
 
+    #[tokio::test]
+    async fn test_insert_and_load_cfd_by_order_id_multiple() {
+        let pool = setup_test_db().await;
+        let mut conn = pool.acquire().await.unwrap();
+
+        let mut order = Order::from_default_with_price(Usd(dec!(10000)), Origin::Theirs).unwrap();
+        order.id = OrderId(Uuid::parse_str("ab00b6f1-550d-4c54-b063-c4b852b96af5").unwrap());
+
+        let cfd = Cfd::new(
+            order.clone(),
+            Usd(dec!(1000)),
+            CfdState::OutgoingOrderRequest {
+                common: CfdStateCommon {
+                    transition_timestamp: SystemTime::now(),
+                },
+            },
+        );
+
+        let order_id = order.id;
+
+        insert_order(&order, &mut conn).await.unwrap();
+        insert_cfd(cfd.clone(), &mut conn).await.unwrap();
+
+        let cfd_from_db = load_cfd_by_order_id(order_id, &mut conn).await.unwrap();
+        assert_eq!(cfd, cfd_from_db);
+
+        // insert second cfd with order-id that is higher in alphabet
```

Comment:
You gotta be joking ...

Reply:
when I saw your message I actually thought - yeah... that one needs more validation. 
It is actually a more general error. The UUID does not matter (I just had used the failing ones to reproduce it). I just flipped the UUIDs and it still works. Will see to generalize the test on Monday and remove the comment!

Reply:
One thing that I find surprising: sqlx does not complain when `fetch_one` returns more than one.

Reply:
Given `fetch_one` does not fail, shall we better not use it then?

Reply:
The docs are somewhat ambiguous here. It sort of suggests that it fails if there are more than 2 results but also doesn't say that clearly anywhere.

Best to open an issue!

Not sure if we need to entirely avoid it but it is good to keep the behaviour in mind. If we do decide that we don't want it, I want it enforced via https://rust-lang.github.io/rust-clippy/rust-1.55.0/index.html#disallowed_method.

Reply:
I had another look, and to be fair the documentation states:

> Errors if no rows were returned. **Extra rows are ignored.** Aggregate queries, use this.

https://docs.rs/sqlx/0.5.1/sqlx/macro.query.html

This means we should probably avoid using it. I'd still go for merge now - we can re-visit this for refactoring later. Better get the fix in :)

Reply:
@da-kami perhaps we could go in for merge now and create an issue to remove `fetch_one()` from the code?